### DISCUSSION TITLE
Wallet Connct through Smart Contract Account

### DISCUSSION THREAD

**Discussion by AbhishekCDN** [alchemyplatform/aa-sdk#940] [Issues]
I have a Smart Contract Account (SCA) through Alchemy Modular Account. Now I want that SCA to connect to a DAAP through Wallet Connect. Do you know how I can do this? We are creating a Wallet Application in which we are creating SCA through Modular Account Alchemy. Now we want to integrate Wallet Connct in it.

**Comment by avarobinson** [comment]
Hi @AbhishekCDN! A few questions:
- Who is the signer provider/owner on your modular account? 
- Did you use the alchemy signer for social login and creation or an existing EOA?
- What's your goal for using WalletConnect? 

If you want to allow users to login with their Modular Account or an existing EOA, you can actually use our [pre-built UI components](https://aa-sdk-site-alpha.vercel.app/react/quickstart?_vercel_share=1H4QaNdwBWvMQImVSV9VJ0HCyD9J6Msl) which support both web2 login via email + passkeys or using an injected wallet like metamask.

**Reply by AbhishekCDN** [reply]
Hey @avarobinson 
1. Who is the signer provider/owner on your modular account?
Ans. -> We have created an EOA through DFNS which is the Siigner in our modular account.

2. Did you use the alchemy signer for social login and creation or an existing EOA?
Ans. -> No, We don't use social login.

3. What's your goal for using WalletConnect?
Ans. -> Suppose we create a SCA modular account in your wallet app. Now user want do a transaction on the Ethereum Chain through Explorer. So in that case we have to connect  our SCA with your app through wallet connect.

**Comment by AbhishekCDN** [comment]
@avarobinson Any Update.
